operator/validator/controller.go

```diff
@@ -456,13 +459,18 @@ func (c *controller) StartValidators() {
 		return
 	}
 
+	mySubnets := c.selfSubnets()
+
 	var ownShares []*ssvtypes.SSVShare
-	var allPubKeys = make([][]byte, 0, len(shares))
+	var pubKeysToFetch [][]byte
 	for _, share := range shares {
 		if c.operatorDataStore.GetOperatorID() != 0 && share.BelongsToOperator(c.operatorDataStore.GetOperatorID()) {
 			ownShares = append(ownShares, share)
 		}
-		allPubKeys = append(allPubKeys, share.ValidatorPubKey[:])
+		subnet := networkcommons.CommitteeSubnet(share.CommitteeID())
+		if uint64(len(mySubnets)) >= subnet && mySubnets[subnet] != 0 {
```

Comment:
I don't think the len check is necessary here because you initialize the mySubnets slice in the same size each time..